hardware_layer/hardware_arduino/firmware/CMakeLists.txt

```diff
@@ -8,28 +8,9 @@ set(CMAKE_CXX_FLAGS "-std=gnu++11 ${CMAKE_CXX_FLAGS}")
 # Remove this if using an Arduino without native USB (eg, other than Leonardo)
 add_definitions(-DUSB_CON)
 
-#====================================================================#
-# Advanced static library
-#====================================================================#
-generate_arduino_library(thruster_lib
-    SRCS ../src/Thruster.cpp
-    HDRS ../include/Thruster.h
-    BOARD uno
-)
-
-generate_arduino_library(pressure_sensor_lib
-    SRCS ../src/MS5837.cpp
-    HDRS ../include/MS5837.h
-    BOARD uno
-)
-
-#====================================================================#
-# Advanced firwmare
-#====================================================================#
 generate_arduino_firmware(arduino_node
-  SRCS arduino_node.cpp ${ROS_LIB_DIR}/time.cpp
-  HDRS ../include/ArduinoConfig.h
-  LIBS thruster_lib pressure_sensor_lib
+  SRCS arduino_node.cpp ${ROS_LIB_DIR}/time.cpp ../src/MS5837.cpp ../src/ArduinoThrust.cpp ../src/ESC.cpp
```

Comment:
Create a static library instead of this (the way it was earlier). It is much cleaner to break into libraries than doing what you have done.